### PR TITLE
support enum type

### PR DESCRIPTION
**Pull request by siddontang**
```
create table t (c enum('a', 'b', 'c'))
```

We will support set type later

### CODE REVIEW

File: mysqldef/enum.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		if strings.EqualFold(n, name) {
+			return Enum{Name: n, Value: uint64(i) + 1}, nil
+		}
+	}
+	return Enum{}, errors.Errorf("item %s is not in enum %v", name, elems)
+}
+
+// ParseEnumValue creates a Enum with special number.
+func ParseEnumValue(elems []string, number uint64) (Enum, error) {
+	if number <= 0 || number > uint64(len(elems)) {
```
Comment (qiuyesuifeng) [inline mysqldef/enum.go]: Use `number == 0` instead of  `number <= 0`?

File: tidb_test.go [hunk 2]
```diff
@@ -894,6 +894,24 @@ func (s *testSessionSuite) TestBootstrap(c *C) {
 	mustExecSQL(c, se, "USE test;")
 }
 
+func (s *testSessionSuite) TestEnum(c *C) {
+	store := newStore(c, s.dbName)
+	se := newSession(c, store, s.dbName)
+
+	mustExecSQL(c, se, "drop table if exists t")
+	mustExecSQL(c, se, "create table t (c enum('a', 'b', 'c'))")
+	mustExecSQL(c, se, "insert into t values ('a'), (2), ('c')")
```
Comment (qiuyesuifeng) [inline tidb_test.go]: How about `insert into t values ('2'),(),(NULL)`?

Comment (siddontang) [inline tidb_test.go]: The same for other types, if column allows null, insert null, if not, fail.

**Comment by shenli** [comment]
Need to update tidb-server? I am pretty sure that there are a few things to do for tidb-server.

**Comment by siddontang** [comment]
No need, we only need to consider it in driver which tidb-server uses.

**Comment by siddontang** [comment]
@shenli  PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**